tiledb/stdx/synchronized_optional/test/CMakeLists.txt

```diff
@@ -30,4 +30,5 @@ commence(unit_test synchronized_optional)
       unit_synchronized_optional.cc
       unit_synchronized_optional_comparisons.cc
       unit_synchronized_optional_swap.cc)
+  this_target_link_libraries(Threads::Threads)
```

Comment:
Note that there is no object library we can link to, because `synchronized_optional` is a header-only library.

Also it should be noted that since its introduction almost two years ago, the only use of `synchronized_optional` is in tests.